### PR TITLE
feat:deduplicate QA

### LINKED ISSUE #431: Data processing - deduplicate Q
1. 对大模型生成的 QA 对进行相似度分析，允许用户手动处理（删除、合并等）
2. 提高 Q  的覆盖率，增大召回率（避免有的内容没有被 Q 覆盖），保证 Indexing 阶段可以覆盖所有文本内容、切片

### PR DESCRIPTION
**Pull request by wangxinbiao**
#### What this PR does / why we need it

feat: Remove duplicate QA data.
chore: Optimized code for chunk and handle sections.

#### Which issue(s) this PR fixes

Fixes #431

### CODE REVIEW

**Comment by bjwswang** [comment]
@wangxinbiao please work with frontend on this enhancement.

File: pypi/data-processing/src/document_chunks/spacy_text_splitter.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Copyright 2023 KubeAGI.
```
Comment (bjwswang) [inline pypi/data-processing/src/document_chunks/spacy_text_splitter.py:1]: 2023 v-> 2024

**Review by bjwswang — CHANGES_REQUESTED** [review]
see the comments

File: pypi/data-processing/src/document_chunks/spacy_text_splitter.py (hunk 1 above)
Comment (wangxinbiao) [reply pypi/data-processing/src/document_chunks/spacy_text_splitter.py]: Corrected

**Review by bjwswang — APPROVED** [review]

**Pull request merged**